plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts

```diff
@@ -163,44 +186,96 @@ export class WinstonInstrumentation extends InstrumentationBase {
     return (original: Winston2LogMethod) => {
       const instrumentation = this;
       return function patchedLog(
-        this: unknown,
+        this: never,
         ...args: Parameters<typeof original>
       ) {
-        const span = trace.getSpan(context.active());
-
-        if (!span) {
-          return original.apply(this, args);
-        }
-
-        const spanContext = span.spanContext();
+        let record: Record<string, any> = {};
+        const config = instrumentation.getConfig();
+        if (!config.disableLogCorrelation) {
+          const span = trace.getSpan(context.active());
+          if (span) {
+            const spanContext = span.spanContext();
+            if (isSpanContextValid(spanContext)) {
+              record = injectRecord(spanContext, record);
+            }
+            instrumentation._callHook(span, record);
 
-        if (!isSpanContextValid(spanContext)) {
-          return original.apply(this, args);
-        }
+            // Inject in metadata argument
+            let isDataInjected = false;
+            for (let i = args.length - 1; i >= 0; i--) {
+              if (typeof args[i] === 'object') {
+                args[i] = Object.assign(args[i], record);
+                isDataInjected = true;
+                break;
+              }
+            }
+            if (!isDataInjected) {
+              const insertAt =
+                typeof args[args.length - 1] === 'function'
+                  ? args.length - 1
+                  : args.length;
 
-        for (let i = args.length - 1; i >= 0; i--) {
-          if (typeof args[i] === 'object') {
-            const record = args[i];
-            injectRecord(spanContext, record);
-            instrumentation._callHook(span, record);
-            return original.apply(this, args);
+              args.splice(insertAt, 0, record);
+            }
           }
         }
 
-        const record = injectRecord(spanContext);
+        if (!config.disableLogSending) {
+          const logRecord: Record<string, any> = {};
+          if (args.length >= 1) {
+            // Level is always first argument
+            logRecord['level'] = args[0];
 
-        const insertAt =
-          typeof args[args.length - 1] === 'function'
-            ? args.length - 1
-            : args.length;
+            // Get meta if present
+            for (let i = args.length - 1; i >= 0; i--) {
+              if (typeof args[i] === 'object') {
+                logRecord['meta'] = args[i];
+                break;
+              }
+            }
 
-        args.splice(insertAt, 0, record);
-        instrumentation._callHook(span, record);
+            let msgArguments = 0;
+            for (let i = 1; i < args.length; i++) {
+              if (typeof args[i] === 'string') {
+                msgArguments++;
+              }
+            }
```

Comment:
nit: You can transform this code to a no-side-effects, shorter version like so:
```suggestion
const msgArguments = args.slice(1).filter(a => typeof a === 'string').length
```

or (if the first argument is always string):
```suggestion
const msgArguments = args.filter(a => typeof a === 'string').length - 1
```

Reply:
I updated this code a little bit, so it doesn't apply anymore, let me know if some other code can be transformed as well.